Gems/OpenXRVk/Code/Source/OpenXRVkDevice.cpp

```diff
@@ -91,20 +91,18 @@ namespace OpenXRVk
         }
 
         {
-#if AZ_TRAIT_OS_IS_HOST_OS_PLATFORM
             VkPhysicalDevice xrVkPhysicalDevice = xrVkInstance->GetActivePhysicalDevice();
-#else
-            // NOTE: When passing a physical device to Vulkan glad loader it uses 'vkEnumerateDeviceExtensionProperties'
-            // to obtain device's extension, but that list doesn't match with the list returned by 'xrGetVulkanDeviceExtensionsKHR'
-            // which is used for OpenXR. This discrepancy results in the context indicating some extensions are available when they
-            // are really not supported in an OpenXR environment. To surpass this issue we're not passing the physical device.
-            VkPhysicalDevice xrVkPhysicalDevice = VK_NULL_HANDLE;
-#endif
-
             // Now that we have created the device, load the function pointers for it.
             const bool functionsLoaded = xrVkInstance->GetFunctionLoader().LoadProcAddresses(
                 &xrVkInstance->GetContext(), xrVkInstance->GetNativeInstance(), xrVkPhysicalDevice, m_xrVkDevice);
             m_context = xrVkInstance->GetContext();
+            // In some cases (like when running with the GPU profiler on Quest2) the extension is reported as available
+            // but the function pointers do not load. Disable the extension if that's the case.
+            if (m_context.EXT_debug_utils && !m_context.CmdBeginDebugUtilsLabelEXT)
+            {
+                m_context.EXT_debug_utils = 0;
+            }
```

Comment:
The instance's context needs to be updated as well. Suggesting:

````
            // Now that we have created the device, load the function pointers for it.
            auto& instanceContext = xrVkInstance->GetContext();
            const bool functionsLoaded = xrVkInstance->GetFunctionLoader().LoadProcAddresses(
                &instanceContext, xrVkInstance->GetNativeInstance(), xrVkPhysicalDevice, m_xrVkDevice);
            // In some cases (like when running with the GPU profiler on Quest2) the extension is reported as available
            // but the function pointers do not load. Disable the extension if that's the case.
            if (instanceContext.EXT_debug_utils && !instanceContext.CmdBeginDebugUtilsLabelEXT)
            {
                instanceContext.EXT_debug_utils = 0;
            }
            m_context = instanceContext;
````

Reply:
This `LoadProcAddresses` call is using instance's context at the moment, if intance and device don't need to have their context with the same data then please use the device's context directly.

But test it because I remember that crashed.

Reply:
Great finding. Good job.